src/is/L42/common/EndError.java

```diff
@@ -5,27 +5,39 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Supplier;
 
 import is.L42.generated.C;
 import is.L42.generated.P;
 import is.L42.generated.Pos;
 
 public abstract class EndError extends RuntimeException{
-  public EndError(List<Pos> poss,String msg){
-    //assert poss!=null: msg;
+  public EndError(List<Pos> poss,Supplier<String> msg){
     assert msg!=null;
-    if(!msg.isEmpty() && !msg.endsWith("\n")){msg+="\n";}
-    this.msgPart=msg;
+    this.msgPartSupplier=()->{String m = msg.get(); return (!m.isEmpty() && !m.endsWith("\n")) ? m + '\n' : m; };
     this.poss=poss;
     }
-  public String msgPart(){return msgPart;}
+  public EndError(List<Pos> poss,String msg){
+    //assert poss!=null;
+    assert msg!=null: msg;
+    if(!msg.isEmpty() && !msg.endsWith("\n")) { msg += "\n"; }
+    this.msgPartString = msg;
+    this.msgPartSupplier = null;
+    this.poss=poss;
+    }
+  public String msgPart(){
+    if(msgPartString == null)
+      msgPartString = msgPartSupplier.get();
+    return msgPartString;
+    }
   @Override public String getMessage(){
-    if(msg==null){msg=ErrMsg.posString(poss)+msgPart+computeMsg();}
+    if(msg==null){msg=ErrMsg.posString(poss)+this.msgPart()+computeMsg();}

```

Comment:
> is anyone calling msgPart() outside of our control? 

Yes, once, MdfTypeSystem.java:135

>What is the use of computeMsg() in the rest of the code base? 

According to your comment, it's an 'override handler'. It allows subclassess of EndError to append things to the message. As far as I can understand, this is never used. It could probably be removed but it's beyond the scope of this MR.